main.cpp

```diff
@@ -86,8 +76,17 @@ struct FloatType
     FloatType& multiply(float myNumber);
     FloatType& divide(float myNumber);
 
+    FloatType& pow(float myNumber);
+    FloatType& pow(const IntType& myNumber);
+    FloatType& pow(const FloatType& myNumber);
+    FloatType& pow(const DoubleType& myNumber);
+
 private:
     float* value = nullptr;
+    FloatType& powInternal(float myNumber);
+    FloatType& powInternal(const IntType& myNumber);
+    FloatType& powInternal(const FloatType& myNumber);
+    FloatType& powInternal(const DoubleType& myNumber);
```

Comment:
There should only be one powInternal function

- [ ] remove these extra `powInternal` functions